### PR TITLE
telemetry refactor, added example for copyaction

### LINKED ISSUE #58: Add telemetry events for relevant actions in app
Dependent on #130

..and now #180

### PR DESCRIPTION
**Pull request by sashei**

### CODE REVIEW

File: lockbox-ios/Action/TelemetryAction.swift [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        telemetryConfig.isCollectionEnabled = false
+        telemetryConfig.isUploadEnabled = false
+        telemetryConfig.updateChannel = "debug"
+#else
+        telemetryConfig.isCollectionEnabled = true
+        telemetryConfig.isUploadEnabled = true
+        telemetryConfig.updateChannel = "release"
+#endif
+
+        // todo: get telemetry-ios PR merged so that we can have a custom PingBuilder for Lockbox
```
Comment (joeyg) [inline lockbox-ios/Action/TelemetryAction.swift]: oops forgot to "open" those classes in the other PR.

File: lockbox-ios/Action/TelemetryAction.swift [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+enum TelemetryEventMethod: String {
+    case tap, signin, startup, foreground, background
+}
+
+enum TelemetryEventObject: String {
+    case app = "app"
+    case entryList = "entry_list"
+    case settingsButton = "settings_button"
```
Comment (joeyg) [inline lockbox-ios/Action/TelemetryAction.swift]: Is there another Issue created for adding in the rest of these events?

Comment (sashei) [reply lockbox-ios/Action/TelemetryAction.swift]: I think that'll be #58, but you have to fight @irrationalagent for them ;)

**Comment by sashei** [comment]
@irrationalagent let me know if you have any questions about how this works... I lied about the "one file" piece, I think you will need to do what I did in `CopyAction.swift` with each of the `Action`s that you'd like to collect telemetry on. They are all in the same folder so shouldn't be hard to find.

**Comment by devinreams** [comment]
@sashei did you want and expect @irrationalagent to continue work for all of #58 on this branch? or do you just want review and questions here and a separate branch for the remaining events?

(in other words: will you merge this soon-ish assuming no concerns?)

**Comment by irrationalagent** [comment]
@devinreams I can do either, whatever's easiest for @sashei

**Comment by sashei** [comment]
@devinreams @irrationalagent that's a good idea -- I was planning to merge this branch when I got an r+ but it would also make sense for Leif to finish the events here.

File: lockbox-ios/Action/CopyAction.swift [hunk 3]
```diff
@@ -43,7 +71,16 @@ class CopyActionHandler: ActionHandler {
...
 
         self.pasteboard.setItems([[UIPasteboardTypeAutomatic: action.text]],
-                                 options: [UIPasteboardOption.expirationDate: expireDate])
-        self.dispatcher.dispatch(action: CopyConfirmationDisplayAction(fieldName: action.fieldName))
+                options: [UIPasteboardOption.expirationDate: expireDate])
+
+        var fieldName: String
+        switch action.field {
+        case .password: fieldName = Constant.string.password
+        case .username: fieldName = Constant.string.username
```
Comment (jhugman) [inline lockbox-ios/Action/CopyAction.swift]: This seems… weird, though happy to be schooled. 

A display string in an `ActionHandler`. Is that right?

File: lockbox-ios/Presenter/RootPresenter.swift [hunk 4]
```diff
@@ -217,7 +225,15 @@ class RootPresenter {
...
-    private func unlockBlindly() {
+    fileprivate func startTelemetry() {
+        Observable.combineLatest(self.telemetryStore.telemetryFilter, self.userDefaults.onRecordUsageData)
+                .filter { $0.1 }
+                .map { $0.0 }
+                .bind(to: self.telemetryActionHandler.telemetryActionListener)
+                .disposed(by: self.disposeBag)
+    }
+
+    fileprivate func unlockBlindly() {
```
Comment (jhugman) [inline lockbox-ios/Presenter/RootPresenter.swift]: Nit: naming/documentation. What does this do?

**Comment by devinreams** [comment]
>maybe merging first is a good idea. I've remembered I've got changes to the doc in progress on another branch off master, so once you merge this I'll rebase then start another PR

Agreed as a team on this earlier today. 👍 

@jhugman will do the code review before merge 🔎

File: lockbox-ios/Presenter/RootPresenter.swift (hunk 4 above)
Comment (sashei) [reply lockbox-ios/Presenter/RootPresenter.swift]: This is a holdover method that keeps the Javascript datastore "unlocked" (i.e.) unenecrypted when the app is open.

**Pull request merged**